reader.go

```diff
@@ -30,48 +31,75 @@ func (o *ReaderOptions) norm(name string) {
 
 // Reader reads data from a remote feed.
 type Reader struct {
-	remote *bfs.Object
-	opt    ReaderOptions
-	ctx    context.Context
-	num    int
+	ctx context.Context
+	opt ReaderOptions
 
-	br io.ReadCloser // bfs reader
-	cr io.ReadCloser // compression reader
-	fd FormatDecoder
+	remotes []*bfs.Object
+	cur     *streamReader
+	fd      FormatDecoder
+
+	pos, num int
 }
 
 // NewReader inits a new reader.
 func NewReader(ctx context.Context, remote *bfs.Object, opt *ReaderOptions) (*Reader, error) {
+	return MultiReader(ctx, []*bfs.Object{remote}, opt), nil
+}
+
+// MultiReader inits a new reader for multiple remotes.  Remotes are read sequentially as if concatenated.
+// Once all remotes are fully read, Read will return EOF.
+func MultiReader(ctx context.Context, remotes []*bfs.Object, opt *ReaderOptions) *Reader {
 	var o ReaderOptions
 	if opt != nil {
 		o = *opt
 	}
-	o.norm(remote.Name())
+	if len(remotes) > 0 {
+		o.norm(remotes[0].Name())
```

Comment:
Super-nit: maybe mention this in comment - that all remotes must be in a single format.

Though, it's kinda expected/obvious/common-sense.